apiTest01

Jamesxj4 opened the pull request:
Intro to Rest Assured

Jamesxj4 committed:
apiTest01

Intro to Rest Assured

Berichicko commented:
**Step 1**

Add KoelTests in TestNG

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/78a741da-98dc-44bc-959e-ac89d5421ce7)

**Step 2**

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/51fb405c-b1be-4e35-bad8-c9530b28dd9f)

or 

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/5dffadbd-4c1b-43d3-9733-71835a1f0b16)


![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/94b84c0a-b04e-46f2-94f9-85c34e2be20c)

**If tests are not running using TestNG, then here is the solution**:

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/9ccc1b42-1b4f-4704-9ede-64fdbc8b01d9)

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/ebb95065-6418-4ca5-829a-a557c3914a6b)

Jamesxj4 commented:
Great, thank you very much !

Jamesxj4 committed:
apiTest

Playlist

Jamesxj4 committed:
Update KoelTests.java

Test failed with
For input string: "public"
java.lang.NumberFormatException: For input string: "public"

io.github.bonigarcia.wdm.config.WebDriverManagerException: java.lang.NumberFormatException: For input string: "public"

Jamesxj4 commented:
the test failed with the following issue, which were test passed yesterday (both playlist and likesong)

io.github.bonigarcia.wdm.config.WebDriverManagerException: java.lang.NumberFormatException: For input string: "public"

For input string: "public"
java.lang.NumberFormatException: For input string: "public"

I closed intellij and reopen after got this issue.  And found out the work of (@Test likesong ) not saved, this one is only typed in but not able to test yet

Berichicko commented:
> the test failed with the following issue, which were test passed yesterday (both playlist and likesong)
> 
> io.github.bonigarcia.wdm.config.WebDriverManagerException: java.lang.NumberFormatException: For input string: "public"
> 
> For input string: "public" java.lang.NumberFormatException: For input string: "public"
> 
> I closed intellij and reopen after got this issue. And found out the work of (@test likesong ) not saved, this one is only typed in but not able to test yet

Hello

Please, re-submit for review

Jamesxj4 commented:
[
![Screenshot 2024-03-22 111658](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/154647121/6f659f20-38ea-4429-b14c-69d65d12fd2f)
](url)

![Screenshot 2024-03-22 111737](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/154647121/4d5659e6-7e37-4928-8bb8-0d3c5f65df66)

Jamesxj4 commented:
the test passed after update to webdrivermanager:5.7.0 and delete the delete the .cache

Berichicko commented:
Hi

I run tests


https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/6fed5cf3-142f-4a89-bc67-51de3431a249

Try it:

**Step 1**

**Select File | Invalidate Caches from the main menu:**


![image](https://github.com/Test-Pro-Bootcamp/A55-QA-Automation-Framework/assets/104761482/334be776-4810-40d1-98cd-ad3b02f2b753)


**Step 2**

**In the Invalidate Caches dialog, you can select :**

- Clear file system cache and Local History: remove the virtual file system cache together with the information stored in Local History.This action might be helpful for troubleshooting purposes when the usual cache invalidation is not enough to solve the problem.

- Clear VCS Log caches and indexes: remove the cache and indexes of the VCS Log.Enable this option if you are certain that your VCS Log indexes are broken, for example, if the Log search doesn't work correctly.

- Mark downloaded shared indexes as broken: do not use the already downloaded shared indexes. After invalidation, IntelliJ IDEA will download and use new indexes if they are available. Otherwise, new indexes will be built locally.

![image](https://github.com/Test-Pro-Bootcamp/A55-QA-Automation-Framework/assets/104761482/e41a2dc7-0c52-4d1b-9929-f4fe2c663669)

Jamesxj4 committed:
sprint12_0322

likeOrUnlikeSong

Jamesxj4 committed:
sprint12_0322v2

like a song
multiple song

Jamesxj4 commented:
Hello, here comes an new issue with api automation testing for Like multiple songs
 
It works fine for (Like or unlike a song): POST api/interaction/like with given().params("songs", "f25b26bc2963e2cb5f4a70511037c0a1")
 
but not for multiple songs (POST api/interaction/batch/like). 
 
How do I use the body parameters with array for automation testing in IntelliJ?
![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/154647121/719a1d66-0af6-4d4e-8b80-eb21c1fff488)

Jamesxj4 committed:
Sprint12_0324

likeMultipleSongs

Berichicko commented:
One possible reason could be that RestAssured is not able to automatically follow the redirection due to certain constraints.

In such cases, you can manually extract the redirection URL from the response headers and then make a separate request to that URL. Here's how you can modify your test method to handle the redirection manually:

```
@Test
public void likeMultipleSongs() {
    String[] multipleSongs = {"f25b26bc2963e2cb5f4a70511037c0a1", "30c13696a68d7794b5468ee3bc9a3fa6"};

    Response response = given()
            .contentType(ContentType.JSON)
            .body(multipleSongs)
            .spec(requestSpec)
            .when()
            .post("https://qa.koel.app/api/interaction/batch/like")
            .then()
            .log().all() // Log all details of the response
            .statusCode(302) // Expecting a redirection status code
            .extract()
            .response();

    // Extract the redirection URL from the response headers
    String redirectionUrl = response.getHeader("Location");

    // Make a separate request to the redirection URL
    given()
            .redirects().follow(false) // Disable automatic redirection
            .when()
            .get(redirectionUrl)
            .then()
            .statusCode(200) // Expecting a successful response after redirection
            .log().all(); // Log all details of the response
}
```

![image](https://github.com/Jamesxj4/A53-QA-Automation-Framework/assets/104761482/2aabbade-d8d4-4ba3-831c-43d0f7fe104c)

For example https://stackoverflow.com/questions/74333165/how-can-i-turn-on-automatically-follow-redirect-in-rest-assured-java
